src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -528,18 +528,16 @@ def visit_ClassDef(self, node):
 
 # Get a mapping from type string to type object, for cast() function
 def get_data_type_string_object_mappings(
-    to_fill_dict: Optional[Dict[str, Type[DataType]]] = None,
```

Comment:
This function only has one call site which provides `to_fill_dict`, so it does not seem reasonable to be more flexible than that.